app/api/search/metadataAggregations.js

```diff
@@ -140,7 +140,10 @@ const propertyToAggregation = (property, dictionaries, baseQuery) => {
     match =>
```

Comment:
This is such a massive hack and this PR is making it worse - can you at least e.g. ?

// TODO: re-build filter instead of relying on internal structure